presto-main/src/main/java/com/facebook/presto/operator/TableScanOperator.java

```diff
@@ -121,6 +124,23 @@ public TableScanOperator(
         this.types = checkNotNull(types, "types is null");
         this.pageSourceProvider = checkNotNull(pageSourceProvider, "pageSourceManager is null");
         this.columns = ImmutableList.copyOf(checkNotNull(columns, "columns is null"));
+        this.systemMemoryUsageListener = new SystemMemoryUpdater(this.operatorContext);
+    }
+
+    public TableScanOperator(
```

Comment:
Don't add this constructor just for tests. Also, when adding multiple constructors, they should all call through to the same base constructor. Having multiple constructors setting fields and doing validity checks makes it very easy to screw up.